src/components/PageBlocks/MetricBase.vue

```diff
@@ -0,0 +1,125 @@
+<template>
+  <wrap v-bind="$props" v-on="$listeners">
+    <div
+      v-for="(m, i) in options.metrics"
+      :key="i"
+    >
+      <h4
+        class="p-2"
+        :style="genStyle(m.labelStyle)"
+      >
+        {{ m.label }}
+      </h4>
+      <p
+        class="p-2"
+        :style="genStyle(m.valueStyle)"
+      >
+        <span
+          v-for="(v, i) in formatResponse(m, i)"
+          :key="i"
+        >
+          <b>
+            {{ v.label }}
+          </b>
+          <br />
+          {{ v.value }}
+        </span>
+      </p>
+    </div>
+  </wrap>
+</template>
+
+<script>
+import base from './base'
+import numeral from 'numeral'
+import moment from 'moment'
+
+export default {
+  extends: base,
+
+  props: {
+    // Preview mode automatically generates data instead of fetching from the API
+    previewMode: {
+      type: Boolean,
+      required: false,
+      default: false,
+    },
+  },
+
+  data () {
+    return {
+      reports: [],
+    }
+  },
+
+  mounted () {
+    this.update()
+  },
+
+  methods: {
+    /**
+     * Generates style for the given metric as defined in configurator
+     */
+    genStyle (s = {}) {
+      const d = {
+        color: s.color,
+        backgroundColor: s.backgroundColor,
+        fontSize: s.fontSize ? s.fontSize + 'px' : undefined,
+      }
+
+      for (const v of Object.keys(d)) {
+        if (d[v] === undefined) {
+          delete d[v]
+        }
+      }
+      return d
+    },
+
+    /**
+     * Performs some post processing on the provided data
+     */
+    formatResponse (m, i) {
+      if (this.previewMode) {
+        return [{
+          label: this.$t('block.metric.edit.previewMetricLabel'),
+          value: Math.floor(Math.random() * 1000) * 1000,
```

Comment:
what is this?

Reply:
@darh This generates a random number that is used inside the configurator's preview window as the value.

Reply:
ok -- this was not clear, code is in the `..Base` cmpnt.